src/test/scala/chiselTests/AnalogSpec.scala

```diff
@@ -0,0 +1,211 @@
+    }
+  }
+
+  it should "NOT be bindable to a direction" in {
+    a [ChiselException] should be thrownBy {
+      elaborate { new Module {
+        val io = IO(new Bundle {
+          val a = Input(Analog(32.W))
+        })
+      }}
+    }
+    a [ChiselException] should be thrownBy {
+      elaborate { new Module {
+        val io = IO(new Bundle {
+          val a = Output(Analog(32.W))
+        })
+      }}
+    }
+  }
+
+  it should "be flippable" in {
+    elaborate { new Module {
+      val io = IO(new Bundle {
+        val a = Flipped(Analog(32.W))
+      })
+    }}
+  }
+
+  // There is no binding on the type of a memory
+  // Should this be an error?
+  ignore should "NOT be a legal type for Mem" in {
+    a [ChiselException] should be thrownBy {
+      elaborate { new Module {
+        val io = IO(new Bundle {})
+        val mem = Mem(16, Analog(32.W))
+      }}
+    }
+  }
+
+  it should "NOT be bindable to Mem ports" in {
+    a [ChiselException] should be thrownBy {
+      elaborate { new Module {
+        val io = IO(new Bundle {})
+        val mem = Mem(16, Analog(32.W))
+        val port = mem(5.U)
+      }}
+    }
+  }
+
+  // TODO This should probably be caught in Chisel
+  // Also note this relies on executing Firrtl from Chisel directly
+  it should "NOT be connectable to UInts" in {
+    a [Exception] should be thrownBy {
+      runTester { new BasicTester {
+        val uint = Wire(init = 0.U(32.W))
+        val sint = Wire(SInt(32.W))
+        sint := uint
+      }}
+    }
+  }
+
+  it should "work with 2 blackboxes bulk connected" in {
+    assertTesterPasses(new AnalogTester {
+      val mod = Module(new AnalogReaderBlackBox)
+      mod.io.bus <> writer.io.bus
+      check(mod)
+    }, Seq("/AnalogBlackBox.v"))
+  }
+
+  it should "error if any bulk connected more than once" in {
+    a [ChiselException] should be thrownBy {
+      elaborate(new Module {
+        val io = IO(new Bundle {})
+        val wires = List.fill(3)(Wire(Analog(32.W)))
+        wires(0) <> wires(1)
+        wires(0) <> wires(2)
+      })
+    }
+  }
+
+  it should "work with 3 blackboxes attached" in {
+    assertTesterPasses(new AnalogTester {
+      val mods = Seq.fill(2)(Module(new AnalogReaderBlackBox))
+      attach(writer.io.bus, mods(0).io.bus, mods(1).io.bus)
+      mods.foreach(check(_))
+    }, Seq("/AnalogBlackBox.v"))
+  }
+
+  it should "work with 3 blackboxes separately attached via a wire" in {
+    assertTesterPasses(new AnalogTester {
+      val mods = Seq.fill(2)(Module(new AnalogReaderBlackBox))
+      val busWire = Wire(Analog(32.W))
+      attach(busWire, writer.io.bus)
+      attach(busWire, mods(0).io.bus)
+      attach(mods(1).io.bus, busWire)
+      mods.foreach(check(_))
+    }, Seq("/AnalogBlackBox.v"))
+  }
+
+  // This does not currently work in Verilator unless Firrtl does constant prop and dead code
+  // elimination on these wires
+  ignore should "work with intermediate wires attached to each other" in {
```

Comment:
Firrtl doesn't error out because this code can be simulated in most tools and synthesized. However, the emitted Verilog has 
```verilog
`ifdef verilator
  `error "Helpful error message"
`endif
```
So that if you try to compile with Verilator you will get a reasonable error message.